pkg/volume/projected/projected.go

```diff
@@ -354,6 +354,27 @@ func (s *projectedVolumeMounter) collectData(mounterArgs volume.MounterArgs) (ma
 				Mode:   mode,
 				FsUser: mounterArgs.FsUser,
 			}
+
+			// Add service ca to ensure parity with service account token secrets
+			// TODO(marun) - Avoid requesting the configmap for every pod
+			//             - Only add to projected volumes added by BoundServiceAccountTokenVolume
+			//             - Handle when a source with name "service-ca.crt" already exists
```

Comment:
I’m sure this is a stupid question, I hope to learn something: why isn’t a natural way to do this is to change the generated projected volume in https://github.com/openshift/kubernetes/blob/4bbda77e59725f85aef253d522a33d5761027b9b/plugin/pkg/admission/serviceaccount/admission.go#L546 ? At least the `service-ca.crt` file conflict would go away entirely.

Reply:
Answering myself: 

> projected volumes only support local object references

Reply:
Following this train of thought, the old token controller creating secrets that include `service-ca.crt` is still running with BoundServiceAccountTokenVolume enabled. So #724 reproduces the old contents of that file (including the root CAs), and is pretty close to the trade-offs of the old approach, and at least survives bootstrap. OTOH “pretty close to the old approach” is probably a good reason not to do it that way.